Exercises/.cmake/FindKokkos.cmake

```diff
@@ -0,0 +1,44 @@
+# Early return if Kokkos is already set up
+# We do not use Kokkos_FOUND as it is not globally defined
+if (TARGET Kokkos::kokkos)
+    return()
+endif ()
+
+set(SPACK_CXX $ENV{SPACK_CXX})
+if (SPACK_CXX)
+    message("found spack compiler ${SPACK_CXX}")
+    set(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
+    set(ENV{CXX} ${SPACK_CXX})
+endif ()
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(default_build_type "RelWithDebInfo")
+    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING
+            "Choose the type of build, options are: Debug, Release, RelWithDebInfo and MinSizeRel."
+            FORCE)
+endif ()
+
+set(Kokkos_COMMON_SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/dep/Kokkos)
+
+if (NOT KokkosTutorials_FORCE_INTERNAL_Kokkos)
+    find_package(Kokkos CONFIG)
+endif ()
```

Comment:
Why do we need a "force" option?

Reply:
I will describe it. 

In fact, during a previous tutorial one of my student had a problem with its ubuntu where Kokkos was installed using a `.deb` but the resulting `KokkosConfig` is not working because of the way debian scatters files everywhere. However, Kokkos was `FOUND`, resulting in skipping the `fetch_content`, and leaving with a non-working build.

With this option, it is possible to bypass any installed Kokkos in standard path.

Reply:
I think you should make it a proper CMake option then.